framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,140 @@ export class CCTransferCommand extends BaseCCCommand {
+		const recipientExist = await userStore.has(
+			methodContext,
+			userStore.getKey(recipientAddress, tokenID),
+		);
+
+		if (!recipientExist) {
+			if (fee < FEE_CCM_INIT_USER_STORE) {
+				throw new Error('Insufficient fee to initialize user account.');
+			}
+
+			// should replace feeAddress with trsSender address
+			await this._tokenMethod.burn(methodContext, feeAddress, tokenID, amount);
 		}
 
-		// let { recipientAddress } = params;
-		// if (ccm.status !== CCM_STATUS_OK) {
-		// 	recipientAddress = params.senderAddress;
-		// }
-
-		// const canonicalTokenID = await this._tokenMethod.getCanonicalTokenID(
-		// 	methodContext,
-		// 	params.tokenID,
-		// );
-		// const recipientExist = await this._tokenMethod.accountExists(
-		// 	methodContext,
-		// 	params.recipientAddress,
-		// );
-		// let receivedAmount = params.amount;
-		// if (!recipientExist) {
-		// 	const minBalance = this._minBalances.find(mb => mb.tokenID.equals(canonicalTokenID))?.amount;
-		// 	if (!minBalance || minBalance > params.amount) {
-		// 		if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-		// 			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-		// 		}
-		// 		return;
-		// 	}
-		// 	receivedAmount -= minBalance;
-		// 	const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-		// 	if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-		// 		const supplyStore = this.stores.get(SupplyStore);
-		// 		const supply = await supplyStore.get(methodContext, canonicalLocalID);
-		// 		supply.totalSupply -= minBalance;
-		// 		await supplyStore.set(methodContext, canonicalLocalID, supply);
-		// 	}
-		// }
-
-		// if (tokenChainID.equals(ownChainID)) {
-		// 	const escrowStore = this.stores.get(EscrowStore);
-		// 	const escrowKey = Buffer.concat([ccm.sendingChainID, tokenLocalID]);
-		// 	const escrowData = await escrowStore.get(methodContext, escrowKey);
-
-		// 	escrowData.amount -= params.amount;
-		// 	await escrowStore.set(methodContext, escrowKey, escrowData);
-		// }
-
-		// const userStore = this.stores.get(UserStore);
-		// await userStore.addAvailableBalanceWithCreate(
-		// 	methodContext,
-		// 	recipientAddress,
-		// 	canonicalTokenID,
-		// 	receivedAmount,
-		// );
+		try {
+			if (tokenChainID.equals(ownChainID)) {
+				const escrowStore = this.stores.get(EscrowStore);
+				const escrowKey = escrowStore.getKey(sendingChainID, tokenID);
+				const escrowData = await escrowStore.get(methodContext, escrowKey);
+
+				if (escrowData.amount > amount) {
+					escrowData.amount -= amount;
+					await escrowStore.set(methodContext, escrowKey, escrowData);
+				} else if (escrowData.amount < amount) {
+					throw new Error('Insufficient balance in escrow account.');
+				}
+			}
+		} catch (error) {
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+		}
+
+		// TODO: check if this implementation is correct
```

Comment:
Pretty trivial, can we please add the issue number if it is to be resolved in future issue :pray: